src/api_strategy.ts

```diff
@@ -0,0 +1,116 @@
+import { CommonDataModel } from "./api_calls"
+import { Cache } from "./cache";
+import { MapCache } from "./map_cache";
+var _ = require('lodash');
+
+/**
+ * An API strategy is an abstract class
+ * meant to provide interface functionality
+ * when getting a response from an API.
+ * Each strategy implements its own version
+ * of getRawResponse, and the APIStrategy
+ * parent abstract class provides context
+ * to the strategy in the form of apiJSON,
+ * and top level mapping methods to convert
+ * the raw response into a useable common
+ * data model.
+ */
+export abstract class APIStrategy {
+    // The JSON for the API
+    protected apiJSON: any;
+
+    // The unique instance ID for the API,
+    // used for caching
+    protected apiID: number;
+
+    // The static counter used for creating a unique API ID
+    private static apiCounter = 0;
+
+    constructor(apiJSON: any) {
+        this.apiJSON = apiJSON;
+        this.apiID = APIStrategy.apiCounter++;
```

Comment:
You should test the behavior of this if the config/API list is updated.

Reply:
I definitely want to start writing tests for the code, and if it is allowed I would do that over winter break. I would say that for now, when the configuration is updated, a whole new list of APIs are pulled in, and they have new apiIDs so they do not pull out of the old cache. However the old cache currently still exists, taking up unnecessary space in memory, and I want to add a call to clear out the cache when that happens, as well as a version of cache that timestamps its data and clears out stale data past a certain time.